pennylane/pauli/pauli_interface.py

```diff
@@ -72,8 +81,9 @@ def _pw_prefactor_tensor(observable: Tensor):
     raise ValueError(f"Expected a valid Pauli word, got {observable}")
 
 
-@_pauli_word_prefactor.register
-def _pw_prefactor_ham(observable: Hamiltonian):
+@_pauli_word_prefactor.register(Hamiltonian)
+@_pauli_word_prefactor.register(LinearCombination)
```

Comment:
Probably better to just use the `pauli_rep` route used by `Prod` and `SProd`.

Reply:
Also deferring to https://app.shortcut.com/xanaduai/story/59353/upgrade-linearcombination
I think we can actually remove this dispatch all-together now that hamiltonian has a pauli_rep